quanto/library/python/udqmm.py

```diff
@@ -0,0 +1,27 @@
+import torch
+
+
+@torch.library.impl("quanto_py::udqmm", "default")
+def udqmm(
+    input: torch.Tensor,
+    weights: torch.Tensor,
+    scale: torch.Tensor,
+    zeropoint: torch.Tensor,
+    axis: int,
+    bits: int,
+    orig_shape: torch.Size,
+    unpacked_shape: torch.Size,
+) -> torch.Tensor:
+    # we transpose it back, so it is simpler to unpack since we have the pack + transposed weights
+    weights = weights.transpose(0, 1)
+    unpacked_weights = torch.ops.quanto.unpack(weights, bits)
+    # TODO : we should proably add that in unpack with with arg unpacked_shape.
+    # Depends if the weights have been transposed or not
+    # if not transposed, we need to do unpacked_weights[: unpacked_shape[0]]
+    unpacked_weights = unpacked_weights[: unpacked_shape[1]]
+    # transpose back
+    unpacked_weights = unpacked_weights.transpose(0, 1)
```

Comment:
When we call the linear layer, the Qbitstensor gets transposed and we end up with transposed weights in the udqmm function. This is quite troublesome for the `unpacking` step since don't have an variable to indicate (e.g. axis for the `ungroup` ) when we should transpose again our weights to unpack them correctly. Do you have any solution to fix this @dacorvo ? We can maybe add an attribute in `PackedTensor` that tell us if the data was transposed, so that we know how to unpack it. 

This is the reason why the test_mm.py are not passing yet. For the test_qlinear.py (where we have transposed weights), there is no issue because of the fix I did above.

Reply:
When the weights are transposed, the axis is also modified, so all the information should be available to the caller. I am not sure to see what the issue is here, because the flow has been validated in python already.

Reply:
The issue is that the packing/unpacking is not dependent on the `axis`.